### PR TITLE
Security analysis

### PR DESCRIPTION
**Pull request by geofjamg**

### CODE REVIEW

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
When running a security analysis with two contingencies list, the following exception occurs in security analysis server :



security-analysis-server_1     | 2020-10-12 11:17:37.704 ERROR 1 --- [or-http-epoll-3] o.g.s.s.s.SecurityAnalysisWorkerService  : java.lang.IndexOutOfBoundsException: Source emitted more than one item
security-analysis-server_1     | 
security-analysis-server_1     | java.lang.IndexOutOfBoundsException: Source emitted more than one item
security-analysis-server_1     | 	at reactor.core.publisher.MonoSingle$SingleSubscriber.onNext(MonoSingle.java:129) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:704) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:580) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxFlatMap$FlatMapInner.onNext(FluxFlatMap.java:986) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1764) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:241) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:385) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1782) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:144) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxContextStart$ContextStartSubscriber.onNext(FluxContextStart.java:96) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxMapFuseable$MapFuseableConditionalSubscriber.onNext(FluxMapFuseable.java:287) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:330) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1782) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:152) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:136) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:252) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:136) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
security-analysis-server_1     | 	at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:378) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
security-analysis-server_1     | 	at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:373) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
security-analysis-server_1     | 	at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:429) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
security-analysis-server_1     | 	at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:625) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
security-analysis-server_1     | 	at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:96) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[netty-codec-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:792) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]
security-analysis-server_1     | 	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
security-analysis-server_1     | 
security-analysis-server_1     | 2020-10-12 11:17:38.699 ERROR 1 --- [onPool-worker-3] com.powsybl.iidm.xml.NetworkXml          : com.powsybl.commons.exceptions.UncheckedXmlStreamException: javax.xml.stream.XMLStreamException: java.io.IOException: Pipe closed
security-analysis-server_1     | 
security-analysis-server_1     | com.powsybl.commons.exceptions.UncheckedXmlStreamException: javax.xml.stream.XMLStreamException: java.io.IOException: Pipe closed
security-analysis-server_1     | 	at com.powsybl.iidm.xml.NetworkXml.write(NetworkXml.java:304) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.NetworkXml.write(NetworkXml.java:309) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.NetworkXml.lambda$copy$9(NetworkXml.java:649) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426) ~[na:na]
security-analysis-server_1     | 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[na:na]
security-analysis-server_1     | 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[na:na]
security-analysis-server_1     | 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[na:na]
security-analysis-server_1     | 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[na:na]
security-analysis-server_1     | 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) ~[na:na]
security-analysis-server_1     | Caused by: javax.xml.stream.XMLStreamException: java.io.IOException: Pipe closed
security-analysis-server_1     | 	at java.xml/com.sun.xml.internal.stream.writers.XMLStreamWriterImpl.writeEndElement(XMLStreamWriterImpl.java:957) ~[na:na]
security-analysis-server_1     | 	at javanet.staxutils.IndentingXMLStreamWriter.writeEndElement(IndentingXMLStreamWriter.java:230) ~[stax-utils-20070216.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.AbstractIdentifiableXml.write(AbstractIdentifiableXml.java:70) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.SubstationXml.writeSubElements(SubstationXml.java:58) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.SubstationXml.writeSubElements(SubstationXml.java:21) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.AbstractIdentifiableXml.write(AbstractIdentifiableXml.java:68) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.NetworkXml.writeBaseNetwork(NetworkXml.java:274) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	at com.powsybl.iidm.xml.NetworkXml.write(NetworkXml.java:297) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
security-analysis-server_1     | 	... 8 common frames omitted
security-analysis-server_1     | Caused by: java.io.IOException: Pipe closed
security-analysis-server_1     | 	at java.base/java.io.PipedInputStream.checkStateForReceive(PipedInputStream.java:260) ~[na:na]
security-analysis-server_1     | 	at java.base/java.io.PipedInputStream.receive(PipedInputStream.java:201) ~[na:na]
security-analysis-server_1     | 	at java.base/java.io.PipedOutputStream.write(PipedOutputStream.java:122) ~[na:na]
security-analysis-server_1     | 	at java.xml/com.sun.xml.internal.stream.writers.UTF8OutputStreamWriter.write(UTF8OutputStreamWriter.java:92) ~[na:na]
security-analysis-server_1     | 	at java.xml/com.sun.xml.internal.stream.writers.UTF8OutputStreamWriter.write(UTF8OutputStreamWriter.java:127) ~[na:na]
security-analysis-server_1     | 	at java.xml/com.sun.xml.internal.stream.writers.XMLStreamWriterImpl.writeEndElement(XMLStreamWriterImpl.java:945) ~[na:na]
security-analysis-server_1     | 	... 15 common frames omitted
security-analysis-server_1     |

**Comment by geofjamg** [comment]
> When running a security analysis with two contingencies list, the following exception occurs in security analysis server :
> 
> security-analysis-server_1 | 2020-10-12 11:17:37.704 ERROR 1 --- [or-http-epoll-3] o.g.s.s.s.SecurityAnalysisWorkerService : java.lang.IndexOutOfBoundsException: Source emitted more than one item
> security-analysis-server_1 |
> security-analysis-server_1 | java.lang.IndexOutOfBoundsException: Source emitted more than one item
> security-analysis-server_1 | at reactor.core.publisher.MonoSingle$SingleSubscriber.onNext(MonoSingle.java:129) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:704) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:580) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxFlatMap$FlatMapInner.onNext(FluxFlatMap.java:986) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1764) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:241) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:385) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1782) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:144) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxContextStart$ContextStartSubscriber.onNext(FluxContextStart.java:96) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxMapFuseable$MapFuseableConditionalSubscriber.onNext(FluxMapFuseable.java:287) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:330) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1782) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:152) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:136) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:252) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:136) ~[reactor-core-3.3.8.RELEASE.jar:3.3.8.RELEASE]
> security-analysis-server_1 | at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:378) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
> security-analysis-server_1 | at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:373) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
> security-analysis-server_1 | at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:429) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
> security-analysis-server_1 | at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:625) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
> security-analysis-server_1 | at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:96) ~[reactor-netty-0.9.10.RELEASE.jar:0.9.10.RELEASE]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[netty-codec-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:792) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]
> security-analysis-server_1 | at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
> security-analysis-server_1 |
> security-analysis-server_1 | 2020-10-12 11:17:38.699 ERROR 1 --- [onPool-worker-3] com.powsybl.iidm.xml.NetworkXml : com.powsybl.commons.exceptions.UncheckedXmlStreamException: javax.xml.stream.XMLStreamException: java.io.IOException: Pipe closed
> security-analysis-server_1 |
> security-analysis-server_1 | com.powsybl.commons.exceptions.UncheckedXmlStreamException: javax.xml.stream.XMLStreamException: java.io.IOException: Pipe closed
> security-analysis-server_1 | at com.powsybl.iidm.xml.NetworkXml.write(NetworkXml.java:304) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.NetworkXml.write(NetworkXml.java:309) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.NetworkXml.lambda$copy$9(NetworkXml.java:649) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426) ~[na:na]
> security-analysis-server_1 | at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[na:na]
> security-analysis-server_1 | at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[na:na]
> security-analysis-server_1 | at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[na:na]
> security-analysis-server_1 | at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[na:na]
> security-analysis-server_1 | at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) ~[na:na]
> security-analysis-server_1 | Caused by: javax.xml.stream.XMLStreamException: java.io.IOException: Pipe closed
> security-analysis-server_1 | at java.xml/com.sun.xml.internal.stream.writers.XMLStreamWriterImpl.writeEndElement(XMLStreamWriterImpl.java:957) ~[na:na]
> security-analysis-server_1 | at javanet.staxutils.IndentingXMLStreamWriter.writeEndElement(IndentingXMLStreamWriter.java:230) ~[stax-utils-20070216.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.AbstractIdentifiableXml.write(AbstractIdentifiableXml.java:70) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.SubstationXml.writeSubElements(SubstationXml.java:58) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.SubstationXml.writeSubElements(SubstationXml.java:21) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.AbstractIdentifiableXml.write(AbstractIdentifiableXml.java:68) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.NetworkXml.writeBaseNetwork(NetworkXml.java:274) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | at com.powsybl.iidm.xml.NetworkXml.write(NetworkXml.java:297) ~[powsybl-iidm-xml-converter-3.6.0.jar:na]
> security-analysis-server_1 | ... 8 common frames omitted
> security-analysis-server_1 | Caused by: java.io.IOException: Pipe closed
> security-analysis-server_1 | at java.base/java.io.PipedInputStream.checkStateForReceive(PipedInputStream.java:260) ~[na:na]
> security-analysis-server_1 | at java.base/java.io.PipedInputStream.receive(PipedInputStream.java:201) ~[na:na]
> security-analysis-server_1 | at java.base/java.io.PipedOutputStream.write(PipedOutputStream.java:122) ~[na:na]
> security-analysis-server_1 | at java.xml/com.sun.xml.internal.stream.writers.UTF8OutputStreamWriter.write(UTF8OutputStreamWriter.java:92) ~[na:na]
> security-analysis-server_1 | at java.xml/com.sun.xml.internal.stream.writers.UTF8OutputStreamWriter.write(UTF8OutputStreamWriter.java:127) ~[na:na]
> security-analysis-server_1 | at java.xml/com.sun.xml.internal.stream.writers.XMLStreamWriterImpl.writeEndElement(XMLStreamWriterImpl.java:945) ~[na:na]
> security-analysis-server_1 | ... 15 common frames omitted
> security-analysis-server_1 |

Fixed in https://github.com/gridsuite/security-analysis-server/pull/2

File: src/components/contingency-list-selector.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                    <Grid item>
+                        <CheckboxList
+                            values={contingencyListNames}
+                            onChecked={handleChecked}
+                        />
+                    </Grid>
+                    <Grid item>
+                        <Alert variant="standard" severity="info">
+                            <FormattedMessage
+                                id="xContingeniesWillBeSimulated"
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply src/components/contingency-list-selector.js]: Done

File: src/translations/en.json [hunk 2]
```diff
@@ -117,5 +117,23 @@
...
 
-    "viewOverloadsTable": "View overloads table"
+    "viewOverloadsTable": "View overloads table",
+
+    "LoadFlow": "LoadFlow",
+    "SecurityAnalysis": "Security analysis",
+
+    "ContingencyListsSelection": "Contingency lists selection",
+    "Start": "Start",
+    "xContingeniesWillBeSimulated": "{x} contingenies will be simulated",
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply src/translations/en.json]: Done

File: src/translations/fr.json [hunk 3]
```diff
@@ -117,5 +117,23 @@
...
 
-    "viewOverloadsTable": "Afficher la table des surcharges"
+    "viewOverloadsTable": "Afficher la table des surcharges",
+
+    "LoadFlow": "Calcul de répartition",
+    "SecurityAnalysis": "Analyse de sécurité",
+
+    "ContingencyListsSelection": "Sélection des listes de contingences",
+    "Start": "Démarrer",
+    "xContingeniesWillBeSimulated": "{x} contingences seront simulées",
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply src/translations/fr.json]: Done

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**